src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -123,7 +123,7 @@ private LocalDefinition EmitAddress(BoundExpression expression, AddressKind addr
 
                 case BoundKind.ConditionalOperator:
                     var conditional = (BoundConditionalOperator)expression;
-                    if (!conditional.IsByRef)
+                    if (!HasHome(conditional, addressKind != AddressKind.ReadOnly))
```

Comment:
>if [](start = 20, length = 2)

What is this change for?

Reply:
We had a bug here, see testcase TestRefConditionalHomelessBranches in CodeGenRefConditionalOperator  for scenarios

---
In reply to: [138713967](https://github.com/dotnet/roslyn/pull/21982#discussion_r138713967) [](ancestors = 138713967)

Reply:
There was a PROTOTYPE comment somewere asking for tests. I added tests and found a bug, so I fixed it.

---
In reply to: [138749075](https://github.com/dotnet/roslyn/pull/21982#discussion_r138749075) [](ancestors = 138749075,138713967)

Reply:
The bug was not too bad - basically inefficiency of IL. 
In some cases we would create unnecessary temps. If both branches of ternary refer to readonly variables, and we need a writeable ref (like if the ternary is a receiver of a struct method call) we can make just one temp at the top, not on all branches recursively.

---
In reply to: [138749293](https://github.com/dotnet/roslyn/pull/21982#discussion_r138749293) [](ancestors = 138749293,138749075,138713967)